snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -349,6 +371,21 @@ def pull(self, force=False):
         self._runner.pull()
         self.mark_pull_done()
 
+    def check_pull(self):
+        # Check to see if pull needs to be updated
+        state_file = states.get_step_state_file(
+            self.plugin.statedir, steps.PULL)
+
+        # Not all sources support checking for updates
+        with contextlib.suppress(NotImplementedError):
```

Comment:
ok, I see the reason for raising now, still would be nice to have a different exception to avoid over catching (or check the payload) to not swallow exceptions like this.

Reply:
Sure thing.

Reply:
Fixed with a custom exception.